lib/drivers/medtronicDriver.js

```diff
@@ -636,16 +741,26 @@ module.exports = function (config) {
       debug('in processData');
       data.settings.deviceId = data.settings.modelNumber + '-' + serial;
       cfg.builder.setDefaults({ deviceId: data.settings.deviceId});
-      // TODO: get all time changes and most recent event
-      cfg.tzoUtil = new TZOUtil(cfg.timezone, new Date().toISOString(), []);
       progress(0);
-      proc.init(cfg);
+
       proc.processPages(data, function (err, records) {
+        var dateRecords =  _.filter(records, function(event) {return event.jsDate;}); // filter out objects without dates
```

Comment:
`RESULT_DAILY_TOTAL`, `UNKNOWN_6C`, `MODEL_522_RESULTS_TOTALS` and `UNKNOWN_6E` have 2-byte date fields only. `UNABSORBED_INSULIN` doesn't have a date field at all.

None of these are in our data model, so can be ignored for now. `RESULT_DAILY_TOTAL` doesn't necessarily occur at midnight so can't be used as a new-day event.